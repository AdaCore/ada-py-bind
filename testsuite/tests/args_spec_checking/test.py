import gen

p = gen.Shape()
pos = gen.Point()

def show_error(fn):
    print "In show_error:"
    try:
        fn()
    except TypeError as e:
        print "    " + e.message
    else:
        print "    OK"


show_error(lambda: p.translate())
show_error(lambda: p.translate("lol"))
show_error(lambda: p.translate(move=pos, pouet=pos))
show_error(lambda: p.translate(pos))
show_error(lambda: p.translate(move=pos))
show_error(lambda: gen.Shape("loul"))
