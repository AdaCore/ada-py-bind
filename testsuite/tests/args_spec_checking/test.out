In show_error:
    Missing argument move
In show_error:
    Wrong type for argument move: expected <class 'gen.Point'>, got <type 'str'>
In show_error:
    Too many arguments. Expected max 2, got 3
In show_error:
    OK
In show_error:
    OK
